src/docdb/nodes.ts

```diff
@@ -100,18 +106,43 @@ export class DocDBCollectionNode implements INode {
 	}
 
 	async getChildren(): Promise<INode[]> {
-		const collLink: string = this.getCollLink();
-		const parentNode = this;
-		const client = new DocumentClient(this.db.getEndpoint(), { masterKey: this.db.getPrimaryMasterKey() });
-		let documents = await this.listDocuments(collLink, client);
-		return documents.map(document => new DocDBDocumentNode(document.id, parentNode, document));
+		if (!this.children) {
+			const collLink: string = this.getCollLink();
+			const client = new DocumentClient(this.db.getEndpoint(), { masterKey: this.db.getPrimaryMasterKey() });
+			let documents = await this.listDocuments(collLink, client);
+			this.addChildrenFromDocuments(documents);
+		}
+		return this.children ? this.children : [];
```

Comment:
Rather than pushing and popping the loadMore node all the time, never include it in children and instead just add it here:
```
return this.children.concat([this._loadMoreNode])
```

(Obviously you would have to save the loadMore node as a private variable called `_loadMoreNode` for that to work)